astropy/stats/spatial.py

```diff
@@ -102,31 +117,36 @@ def x_max(self, value):
             )
 
     @property
-    def y_min(self):
+    def y_min(self) -> float | None:
         return self._y_min
 
     @y_min.setter
-    def y_min(self, value):
+    def y_min(self, value: float | None) -> None:
         if value is None or isinstance(value, (float, int)):
             self._y_min = value
         else:
             raise ValueError(f"y_min is expected to be a real number. Got {value}.")
 
     @property
-    def x_min(self):
+    def x_min(self) -> float | None:
         return self._x_min
 
     @x_min.setter
-    def x_min(self, value):
+    def x_min(self, value: float | None) -> None:
         if value is None or isinstance(value, (float, int)):
             self._x_min = value
         else:
             raise ValueError(f"x_min is expected to be a real number. Got {value}.")
 
-    def __call__(self, data, radii, mode="none"):
+    def __call__(
+        self,
+        data: NDArray,
+        radii: NDArray,
+        mode: Literal["none", "translation", "ohser", "var-width", "ripley"] = "none",
```

Comment:
This literal is used many times and can be made into a TypeAlias for repeat usage.

Reply:
The longer a type annotation is, the more can be gained by introducing an alias for it. However, it needs to be done in a public `typing` module or else Sphinx will start complaining about it.

Reply:
Aliases that are only used in private function signatures should be safe because Sphinx shouldn't ever try to render them, but if an alias is used in the signature of a public function then it can cause problems (I think).

Reply:
I think that's right, when building the docs I got warnings like `py:class reference target not found: ArrayLike [ref.class]` until I added `py:class ArrayLike` in `docs.nitpick_exceptions`

Since this alias would be used in public function signatures, I don't think it's worth adding it for the 4 cases of `Literal["none", "translation", "ohser", "var-width", "ripley"]`. But if you'd prefer that I do add it, I can and will add the new custom type to `nitpick_exceptions`.

Reply:
Hm. Seems like something Sphinx should have a better fix for. When a type annotation gets very long, as in `sigma_clipping.py` it's helpful to condense using aliases.

@jeffjennings, if you don't mind, could you try a private alias, like `_ModeOps: TypeAlias = Literal["none", "translation", "ohser", "var-width", "ripley"]` ?
Then we'll know whether some consolidation can happen, which might help readability in the return annotations in `sigma_clipping.py`

Reply:
Sure, fixed in [845e89d](https://github.com/astropy/astropy/pull/16562/commits/845e89dc85005a3ec03c3e6af51cd382ecbb91cd)

Reply:
I haven't had time to review this properly, but I will comment on this.

When we have arguments that accept some fixed set of strings as values then the best thing to do might be to use [`StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum), but that is only available starting from Python 3.11 and we are still supporting Python 3.10.

Reply:
Sounds like a good follow-up PR!

Reply:
> Sure, fixed in [845e89d](https://github.com/astropy/astropy/pull/16562/commits/845e89dc85005a3ec03c3e6af51cd382ecbb91cd)

Unfortunately the docs are failing so I can't yet review whether this worked.
Please ping me when the docs are building.

Reply:
Added a comment with TODO for using `StrEnum` in [a7c18f5](https://github.com/astropy/astropy/pull/16562/commits/a7c18f526377c252e4100dbe63446c9a426e8821)

Reply:
@nstarman I don't know what's breaking the docs build, and when I try to build the docs locally, there's no issue until the build hangs at 'embedding documentation hyperlinks...'. 

The output of the CI `docs/readthedocs.org:astropy` [build test](https://app.readthedocs.org/projects/astropy/builds/24750034/) isn't helpful either; the end of the log there says simply "build finished with problems, 3 warnings." I don't think the 'problems' or warnings are logged anywhere? 

Because I pushed multiple commits at once such that the docs build test was only run on the end result, I can only isolate the issue to 6 commits (those on June 17), and I don't see a way to re-run the test on previous commits. So I'm ready to revert these commits and incrementally re-add them to find the issue. Is `git revert` ok to use for Astropy (I assume it should be as it doesn't rewrite the history)? 

Sorry for the hassle, feel free to suggest anything different.

Reply:
The readthedocs build problem is unrelated, I think. The output is indeed annoying, but scrolling up the raw log, one finds:
```
/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/16562/lib/python3.11/site-packages/astropy/time/core.py:docstring of astropy.time.core.TimeBase.ptp:8: WARNING: py:meth reference target not found: numpy.ndarray.ptp [ref.meth]
/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/16562/lib/python3.11/site-packages/astropy/time/core.py:docstring of astropy.time.core.TimeBase.ptp:8: WARNING: py:meth reference target not found: numpy.ndarray.ptp [ref.meth]
/home/docs/checkouts/readthedocs.org/user_builds/astropy/conda/16562/lib/python3.11/site-packages/astropy/time/core.py:docstring of astropy.time.core.TimeBase.ptp:8: WARNING: py:meth reference target not found: numpy.ndarray.ptp [ref.meth]
```
This was related to readthedocs pulling in numpy 2.0 and is solved by #16583 - so, hopefully a rebase will get it all to pass!

Reply:
Ah I didn't realize the warnings were enough to fail the build, I thought the 'problems' in the log referred to full errors. Thank you! I rebased and the docs tests now pass.

@nstarman I reverted the `NDArray[np.floating]` annotations to `NDArray[float]` because `np.floating` was causing docs warnings; I'd like to save that for another day if we could, when types get more standardized across the codebase. (There's one other instance of `NDArray[float]` in cosmology [tests](https://github.com/astropy/astropy/blob/62bd2be1e3d883914998b2880165aa865c318e0f/astropy/cosmology/tests/test_core.py#L36).)